main.go

```diff
@@ -163,35 +159,6 @@ func getUnrestrictedClient() client.Client {
 	return k8sClient
 }
 
-func mapCRDAvailability(crdNames ...string) map[string]bool {
-
-	var options client.Options
-	options.Scheme = runtime.NewScheme()
-	utilruntime.Must(apiextensionsv1.AddToScheme(options.Scheme))
-
-	k8sClient, err := client.New(config.GetConfigOrDie(), options)
-	if err != nil {
-		setupLog.Error(err, "error creating client")
-		os.Exit(1)
-	}
-
-	CRDExist := map[string]bool{}
-
-	for _, crdName := range crdNames {
-		crd := apiextensionsv1.CustomResourceDefinition{}
-		err = k8sClient.Get(context.Background(), types.NamespacedName{Name: crdName}, &crd)
-
-		if err != nil && !errors.IsNotFound(err) {
-			setupLog.Error(err, fmt.Sprintf("Error Getting CRD for %s", crdName))
-			os.Exit(1)
-		}
-
-		CRDExist[crdName] = crd.UID != ""
-	}
-
-	return CRDExist
-}
```

Comment:
Why are we moving this into the util package? On a general note, we should read all CRDs only once!

Reply:
That is exactly the point!

Reply:
Then on what basis we should restart the operator? We need to check if OCS related CRDs are present or not, so moved this function in util to update the list of CRDs when required by the offering reconcilers

Reply:
Just fetch the current state of the existence of the CRD in question and compare it to the information in the CRD map

Reply:
That would be the approach that I will take